lib/assessment.rb

```diff
@@ -43,6 +43,6 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+    filename.each_with_object('') { |line, memo| memo << line.chomp! }
```

Comment:
Close. You're missing the fact that we need to actually load the file, using `File.open(filename)`. Also, don't forget to add back a space after removing the newline.